add imKey Hardware Wallet

makoshan opened the pull request:
Add imKey, a hardware wallet backed by imToken.
Founded in 2019, imKey uses a CC EAL6+ certified secure chip to keep private keys safe during generation and storage.
It works with both imToken and Rabby wallets, and has become a popular choice for users who value security and convenience.

- Official Website: https://imkey.im
- Source Code: https://github.com/consenlabs/imkey-core

makoshan committed:
add imKey Hardware Wallet

makoshan committed:
fix build bugs

makoshan committed:
Fix imKey license type usage

makoshan committed:
Fix imKey license type usage

makoshan committed:
Fix imKey license error

makoshan committed:
Fix imKey license error

makoshan committed:
Fix imKey License import — final version

polymutex commented:
> Hi @polymutex, I just tested the URL https://imkey.im with curl — it returned HTTP 200 OK with a response time of around 0.386 seconds. Everything looks normal.

That's good; but as per the test output, please add the known URL to the test (in this PR):

```
A new valid URL was detected, and needs to be added to this test to avoid re-fetching it on every run.

Add the following to `knownValidUrls`:

	{
		url: 'https://imkey.im/',
		urlHash: 'a48c5da2453ffb4f5e7ba1f3f9c46bd30e1de5b9',
		retrieved: '2025-11-05'
	},
```

makoshan committed:
Fix imKey url test

makoshan committed:
Merge upstream/beta: Resolve conflict in tests/url-check.test.ts

makoshan committed:
fix: resolve ESLint CI errors in TooltipOrAccordion.svelte

Replace ComponentProps<typeof Tooltip> with explicit literal types to fix
@typescript-eslint/no-unsafe-assignment errors. The Tooltip component is
imported after the props type definition, causing TypeScript to infer
'any' types.

polymutex commented:
Don't worry about the CI failure on `TooltipOrAccordion.svelte`, that's not this PR's fault.

PR looks good in general but I haven't actually reviewed the hardware wallet data, that's more @nconsigny's expertise. Note that we are refactoring this data quite a bit still (@maykelxyz is working on the bug bounty program data for example) so there may be further changes.

makoshan committed:
恢复 TooltipOrAccordion 组件使用 ComponentProps 类型
